source/use-cases/sensitive-data-encryption.txt

```diff
@@ -57,12 +57,127 @@ cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+JSON Schema is used to specify several configuration details, allowing the drivers
+to automatically encrypt and decrypt fields. 
+
+* To do so, you need to specify two things in JSON Schema for each encrypted field:
+    a) The encryption type (deterministic or random)
+    b) The data key
+    c) The BSON Type
+* You can use a single data key for all encrypted fields or you can use individual data keys that are specific to each field.
+
+
+The following table represents the fields of data, MedcoMD will be using in their Medical Care Management System. Each field
+will be either unencrypted or encrypted. Encrypted fields will be encrypted through `Deterministic Encryption <https://docs.mongodb.com/manual/core/security-client-side-encryption/#deterministic-encryption>`_
+or `Random encryption <https://docs.mongodb.com/manual/core/security-client-side-encryption/#random-encryption>`_.
+
+.. table:: 
+    :widths: 50 50
+
+    +-----------------------------+---------------------+---------------------+
+    |Field type                   |Encryption Algorithm | BSON Type           |
+    +=============================+=====================+=====================+
+    | Name                        | Non-Encrypted       | String              |
+    +-----------------------------+---------------------+---------------------+
+    | SSN                         | Deterministic       |  Int                |
+    +-----------------------------+---------------------+---------------------+
+    | Blood Type                  | Random              |  String             |
+    +-----------------------------+---------------------+---------------------+
+    | Medical Records             | Random              |  Array              |
+    +-----------------------------+---------------------+---------------------+
+    | Insurance: Policy Number    | Deterministic       |  (Nested) Int       |
+    +-----------------------------+---------------------+---------------------+
+    | Insurance: Provider         | Non-Encrypted       |  (Nested) String    |
+    +-----------------------------+---------------------+---------------------+
+
+
+You will use the secure client, created in an earlier, by passing it a JSON object that contains your schema. 
+In the JSON, you can specify `encryptMetadata <https://docs.mongodb.com/manual/reference/security-client-side-automatic-json-schema/#encryptmetadata-schema-keyword>`_ that is passed down to the fields under the ``properties``
+nested object. 
+
+.. code::
+
+        {
+            "bsonType" : "object",
+            "encryptMetadata" : {
+                "keyId" : // copy and paste your keyID generated here
+            },
+            "properties": {
+                // copy and paste your fields here 
+            }
+        }
+
+For the MedcoMD system, we will create JSON objects for each field and paste them into the properties 
+object of the JSON Schema above.
+
+
+.. code::
```

Comment:
I am suggesting the same format as the one you linked, but maybe it was not clear enough. In my example, here are the fields to which each line belongs:

```
Property name: JSON Schema for Social Security Number (SSN)
```
```
Description: SSN is a 9-digit identification number that can be used to verify a patient's identity.

Note: keyId can be specified, but is omitted because it inherits the property from encryptMetadata ...
```
```
Code example: <Code example>
```

What you currently have:
```
Code comment with a combination of describing deterministic encryption and how it applies to the field:
// Deterministic encryption supports querying...  of the encrypted data.
```

A couple reasons I think formatting it the way I suggested is helpful:
1) By including the description in the code block, it will have to be re-written as a comment in every language example
2) You can use the "Note" directive to separate the description from any other context you want to provide

Reply:
@mongomoe Just wanted to surface this since I haven't seen any responses to this thread.